### ISSUE TITLE
Idle event

### ISSUE THREAD

**Issue by bvssvni** [PistonDevelopers/event_loop#33]
Instead of just going to sleep, it would be nice to have an "idle" event which gets called before going to sleep with the amount of seconds to sleep. When the "idle" event is handled, it sleeps for the remaining time. This could allow background tasks in single thread applications.

The "idle" event should only be activated before sleeping and the poll is empty, and only if it did not emitted "idle" last time.

**Comment by bvssvni** [comment]
This could allow background tasks in single thread applications.

**Comment by bvssvni** [comment]
I wonder if it would allow composing event loops together.

**Comment by bvssvni** [comment]
Composing event loops requires non-block when polling events from another source

**Comment by bvssvni** [comment]
If the event loop never sleeps before the event poll is empty, then it can be unblocked by creating a wrapper that returns and "idle" event whenever the poll is empty.

**Comment by bvssvni** [comment]
The "idle" event should only be activated before sleeping and the poll is empty, and only if it did not emitted "idle" last time.
